polynote-kernel/src/main/scala/polynote/kernel/ScalaCompiler.scala

```diff
@@ -600,9 +593,12 @@ object ScalaCompiler {
     global            <- ZIO(new Global(settings, KernelReporter(settings)))
     counter           <- kernelCounter
     notebookPackage    = s"notebook$counter"
-    classLoader       <- makeClassLoader(settings, dependencyClasspath ++ transitiveClasspath).memoize
+    classLoader       <- makeClassLoader(settings, dependencyClasspath ++ transitiveClasspath)
   } yield new ScalaCompiler(global, notebookPackage, classLoader, dependencyClasspath, otherClasspath)
 
+  def apply(dependencyClasspath: List[File], transitiveClasspath: List[File], otherClasspath: List[File]): RIO[Config with System, ScalaCompiler] =
+    apply(dependencyClasspath, transitiveClasspath, otherClasspath, identity[Settings])
```

Comment:
curious, why not provide a default value? `modifySettings: Settings => Settings = identity[Settings]`